guava/src/com/google/common/math/Quantiles.java

```diff
@@ -603,22 +639,30 @@ private static void movePivotToStartOfSlice(double[] array, int from, int to) {
    * allRequired[i]} for {@code i} in the range [{@code requiredFrom}, {@code requiredTo}]. These
    * indexes must be sorted in the array and must all be in the range [{@code from}, {@code to}].
    */
+  @SuppressWarnings({"lowerbound:argument.type.incompatible",/*(1): Since allRequired is sorted, and requiredBelow < requiredChosen,
+  allRequired[requiredBelow] <= allRequired[requiredChosen], therefore allRequired[requiredBelow] is always <= required. Based on the condition
+  of if and while, allRequired[requiredBelow] != required. Because allRequired[requiredBelow] <= required, we know that allRequired[requiredBelow] < required,
+  since allRequired[requiredBelow] >= requiredFrom >= 0, then required > 0. */
```

Comment:
I don't see why `allRequired[requiredBelow] >= requiredFrom` should hold. Can you simply remove that from the equation and keep just `allRequired[requiredBelow] >= 0`?